contracts/tests/SpigotedLine.t.sol

```diff
@@ -937,5 +937,68 @@ contract SpigotedLineTest is Test {
     function test_can_whitelist_as_arbiter() public {
       assertTrue(line.updateWhitelist(bytes4("0000"), true));
     }
+
+    function test_cant_use_and_repay_if_unauthorized() public {
+      _borrow(line.ids(0), lentAmount);
+      vm.prank(address(1));
+      vm.expectRevert("Unauthorized");
+      line.useAndRepay(1);
+    }
+    
+    
+    function test_lender_can_use_and_repay() public {
+      deal(address(lender), lentAmount + 1 ether);
+      deal(address(revenueToken), MAX_REVENUE);
+      address revenueC = address(0xbeef); // need new spigot for testing
+      bytes32 id = _createCredit(address(revenueToken), Denominations.ETH, revenueC);
+      _borrow(id, lentAmount);
+
+      bytes memory tradeData = abi.encodeWithSignature(
+        'trade(address,address,uint256,uint256)',
+        address(revenueToken),
+        Denominations.ETH,
+        1 gwei,
+        lentAmount
+      );
+
+      hoax(borrower);
+      line.claimAndTrade(address(revenueToken), tradeData);
+
+      vm.prank(lender); // prank lender
+      line.useAndRepay(lentAmount);
+      (, uint256 principal,,,,,) = line.credits(line.ids(0));
+      assertEq(principal, 0);
+    }
+    
+    function test_cant_claim_and_repay_if_unauthorized() public {
```

Comment:
can you also add test that the arbiter cant use/claimAndRepay. Just want to be explicit since they have accessto funds in other scenarios

Reply:
No. We used to have it setup that way to prevent griefing but then we decided it makes more  sense to have the first lender in payment queue be able to call instead of the arbiter

Reply:
done